dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1121,6 +1178,118 @@
 		</ItemGroup>
 	</Target>
 
+	<Target Name="_XamarinComputeIlcCompileInputs">
+		<PropertyGroup>
+			<!-- Ask ILC to produce a static library -->
+			<NativeLib>static</NativeLib>
+		</PropertyGroup>
+
+		<!-- Remove libSystem.Net.Security.Native.a from tvOS when using NativeAOT, it's not shipped on tvOS -->
+		<ItemGroup>
+			<DirectPInvoke Remove="libSystem.Net.Security.Native" />
+			<NetCoreAppNativeLibrary Remove="System.Net.Security.Native" />
+			<NativeLibrary Remove="@(NativeLibrary)" Condition="'%(Filename)' == 'libSystem.Net.Security.Native'" />
+			<LinkerArg Remove="@(LinkerArg)" Condition="'%(Filename)' == 'libSystem.Net.Security.Native'" />
+		</ItemGroup>
+
+		<ItemGroup>
+			<!-- We need to mark all __Internal P/Invokes as direct, so that the native linker doesn't remove them -->
+			<DirectPInvoke Include="__Internal" />
+
+			<!-- Give ILLink's output to ILC -->
+			<_UpdatedManagedAssemblyToLink Include="@(ManagedAssemblyToLink->'$(IntermediateLinkDir)%(Filename)%(Extension)')" Condition="Exists('$(IntermediateLinkDir)%(Filename)%(Extension)')" />
+			<ManagedAssemblyToLink Remove="@(ManagedAssemblyToLink)" />
+			<ManagedAssemblyToLink Include="@(PrivateSdkAssemblies);@(_UpdatedManagedAssemblyToLink)" />
+			<ManagedBinary Include="$(IntermediateLinkDir)$(TargetName)$(TargetExt)" />
+			<IlcCompileInput Include="@(ManagedBinary)" />
+			<IlcReference Include="@(ManagedAssemblyToLink)" Exclude="@(ManagedBinary)" />
+
+			<!-- Process UnmanagedCallersOnly attributes from every assembly -->
+			<UnmanagedEntryPointsAssembly Remove="System.Private.CoreLib" />
+			<UnmanagedEntryPointsAssembly Include="@(_UpdatedManagedAssemblyToLink->'%(Filename)')" />
+
+			<!-- Disable trimming by default (assemblies aren't trimmed unless they're marked IsTrimmable) -->
+			<IlcArg Include="--defaultrooting" />
+
+			<!-- Link in the output from ILC -->
+			<_NativeExecutableObjectFiles Include="$(NativeObject)" />
+		</ItemGroup>
+
+		<ItemGroup>
+			<!--
+
+				ILLink's build logic will add @(IntermediateAssembly) to
+				TrimmerRootAssembly. ILC understands TrimmerRootAssembly, and
+				will root such assemblies. The problem is that
+				TrimmerRootAssembly is documented to be just the assembly name
```

Comment:
The compiler should be quirking this - the quirk is here: https://github.com/dotnet/runtime/blob/74196a07ce50054c998c298eeed29f22f780bbf2/src/coreclr/tools/aot/ILCompiler/Program.cs#L268-L272

I wonder why it doesn't work. If the path is relative, did someone change the current directory?

Reply:
@MichalStrehovsky this happens otherwise:

```
EXEC : error : Module with same simple name already exists /Users/rolf/work/maccore/nativeaot/xamarin-macios/tests/dotnet/MySimpleApp/iOS/obj/Release/net8.0-ios/ios-arm64/MySimpleApp.dll [/Users/rolf/work/maccore/nativeaot/xamarin-macios/tests/dotnet/MySimpleApp/iOS/MySimpleApp.csproj]
  System.IO.FileNotFoundException: Module with same simple name already exists /Users/rolf/work/maccore/nativeaot/xamarin-macios/tests/dotnet/MySimpleApp/iOS/obj/Release/net8.0-ios/ios-arm64/MySimpleApp.dll
     at ILCompiler.CompilerTypeSystemContext.AddModule(String, String, Boolean, CompilerTypeSystemContext.ModuleData, Boolean) + 0x5d0
     at ILCompiler.CompilerTypeSystemContext.GetOrAddModuleFromPath(String, Boolean, Boolean) + 0xcf
     at ILCompiler.Program.Run() + 0x120e
     at ILCompiler.ILCompilerRootCommand.<>c__DisplayClass212_0.<.ctor>b__0(InvocationContext context) + 0x2f3
```

rsp file: https://gist.github.com/rolfbjarne/77c4ceda66c98423f1eda1f2763e389e
binlog: [msbuild.binlog.zip](https://github.com/xamarin/xamarin-macios/files/11815876/msbuild.binlog.zip)

Reply:
There are two different assemblies with the same simple name:

```
obj/Release/net8.0-ios/ios-arm64/linked/MySimpleApp.dll
...
--root:obj/Release/net8.0-ios/ios-arm64/MySimpleApp.dll
```

The rooted one should be the same as the one passed as input.

Reply:
Ah, good catch. That makes the comment wrong, but the code is still correct though.

A better comment would be something like:

```cs
/* We need to adjust TrimmerRootAssembly, because ILLink sets it to @(IntermediateAssembly), but for ILC it must be the output from ILLink and not its input. */
```